### PR TITLE
Refactored schema-generator to froms-shared

### PR DESCRIPTION
**Pull request by skifahrer**
- schema generator (now forms-shared) escaped from next structure to root/
- fixed jsonschemas issue
- changed JSONSchema7 to BAJSONSchema7
- import paths and ts conifgs updated for proper building
- forms-shared now have dockerfile for image building
- updated dockerfile on next to support forms-shared image
- pipelines now build forms-shared first both on validation and deploy

### CODE REVIEW

**Comment by MarekBodingerBA** [comment]
NIT: Can you please rename "shared" to "forms-shared" and "FormsJSONSchema7" to "BAJSONSchema7"?

Also, do we run tests in "forms-shared" in pipeline here, if not, can we? (snapshots need to be regenerated)

**Review by MarekBodingerBA — APPROVED** [review]

File: tests/cypress/e2e/form/formRealEstateTaxReturn.cy.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /// <reference types="cypress" />
 
 const path = require("path");
-const { esbsNationalityCiselnik } = require("../../../../next/schema-generator/src/definitions/priznanie-k-dani-z-nehnutelnosti/esbsCiselniky")
+const { esbsNationalityCiselnik } = require("../../../../shared/src/definitions/priznanie-k-dani-z-nehnutelnosti/esbsCiselniky")
```
Comment (MarekBodingerBA) [inline tests/cypress/e2e/form/formRealEstateTaxReturn.cy.ts:4]: import

File: next/next.config.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ const nextConfig = {
   eslint: {
     dirs: ['components/', 'pages/', 'utils/', 'backend/', 'frontend/'],
   },
+  transpilePackages: ['ui'],
```
Comment (MarekBodingerBA) [inline next/next.config.js:26]: Is this needed?

**Comment by MarekBodingerBA** [comment]
Added two comments, otherwise LGTM.

**Review by mpinter — APPROVED** [review]
Probably good - there's one ts error which probably needs fixing.

Would think again about the naming of forms-shared, but if it works we can move forward and return to this later.

**Pull request merged**